groups/ntc/ntco/ntco_ioring.cpp

```diff
@@ -0,0 +1,6359 @@
+
+    void*       data = source.data();
+    bsl::size_t size = source.size();
+
+    iovecArray[0].iov_base = data;
+    iovecArray[0].iov_len  = size;
+
+    event->d_numBytesAttempted = size;
+
+    message->msg_iov    = iovecArray;
+    message->msg_iovlen = 1;
+
+    const bool specifyEndpoint = !options.endpoint().isNull();
+
+    if (specifyEndpoint) {
+        ::sockaddr_storage* socketAddress =
+            event->address< ::sockaddr_storage>();
+
+        bsl::size_t socketAddressSize;
+        error = ntsu::SocketUtil::encodeEndpoint(socketAddress,
+                                                 &socketAddressSize,
+                                                 options.endpoint().value());
+
+        message->msg_name    = socketAddress;
+        message->msg_namelen = static_cast<socklen_t>(socketAddressSize);
+    }
+
+    d_operation = static_cast<bsl::uint8_t>(ntco::IoRingOperation::e_SENDMSG);
+    d_handle    = handle;
+    d_event     = reinterpret_cast<__u64>(event);
+    d_flags     = 0;
+    d_address   = reinterpret_cast<__u64>(message);
+
+    return ntsa::Error();
+}
+
+ntsa::Error IoRingSubmission::prepareSend(
+    ntcs::Event*                                 event,
+    const bsl::shared_ptr<ntci::ProactorSocket>& socket,
+    ntsa::Handle                                 handle,
+    const ntsa::ConstBuffer&                     source,
+    const ntsa::SendOptions&                     options)
+{
+    ntsa::Error error;
+
+    BSLS_ASSERT(event->d_status == ntcs::EventStatus::e_FREE);
+
+    event->d_type   = ntcs::EventType::e_SEND;
+    event->d_status = ntcs::EventStatus::e_PENDING;
+    event->d_socket = socket;
+
+    ::msghdr* message = event->message< ::msghdr>();
+    bsl::memset(message, 0, sizeof(::msghdr));
+
+    bsl::size_t numBuffersMax;
+    ::iovec*    iovecArray = event->buffers< ::iovec>(&numBuffersMax);
+
+    void*       data = const_cast<void*>(source.data());
+    bsl::size_t size = source.size();
+
+    iovecArray[0].iov_base = data;
+    iovecArray[0].iov_len  = size;
+
+    event->d_numBytesAttempted = size;
+
+    message->msg_iov    = iovecArray;
+    message->msg_iovlen = 1;
+
+    const bool specifyEndpoint = !options.endpoint().isNull();
+
+    if (specifyEndpoint) {
+        ::sockaddr_storage* socketAddress =
+            event->address< ::sockaddr_storage>();
+
+        bsl::size_t socketAddressSize;
+        error = ntsu::SocketUtil::encodeEndpoint(socketAddress,
+                                                 &socketAddressSize,
+                                                 options.endpoint().value());
+
+        message->msg_name    = socketAddress;
+        message->msg_namelen = static_cast<socklen_t>(socketAddressSize);
+    }
+
+    d_operation = static_cast<bsl::uint8_t>(ntco::IoRingOperation::e_SENDMSG);
+    d_handle    = handle;
+    d_event     = reinterpret_cast<__u64>(event);
+    d_flags     = 0;
```

Comment:
Approach to explicitly zero `d_flags` here and in other places does not look consistent.
`IoRingSubmission` members are zeroed in its constructor. And despite that fact, `d_flags` is zeroed here one more time, while other members aren't. I think it's better to remove explicit zeroeing of `d_flags`.

Reply:
OK